dbt_subprojects/hourly_spellbook/models/_sector/perpetual/projects/pika/optimism/pika_v4_optimism_perpetual_trades.sql

```diff
@@ -0,0 +1,171 @@
+{{ config(
+	
+	schema = 'pika_v4_optimism',
+	alias = 'perpetual_trades',
+	partition_by = ['block_month'],
+	materialized = 'incremental',
+	file_format = 'delta',
+	incremental_strategy = 'merge',
+	unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index'],
+    post_hook='{{ expose_spells(\'["optimism"]\',
+                                "project",
+                                "pika",
+                                \'["msilb7", "drethereum", "rplust","princi"]\') }}'
+	)
+}}
+
+{% set project_start_date = '2021-11-22' %}
+
+WITH positions AS (
+	SELECT
+		positionId
+		,user AS user
+		,productId
+		,CAST(isLong AS VARCHAR) AS isLong
+		,price
+		,oraclePrice
+		,margin
+		,leverage
+		,fee
+		,contract_address
+		,evt_tx_hash
+		,evt_index
+		,evt_block_time
+		,evt_block_number
+		,'4' AS version
+	FROM {{ source('pika_protocol_v4_optimism', 'PikaPerpV4_evt_NewPosition') }}
+	{% if is_incremental() %}
+	WHERE evt_block_time >= DATE_TRUNC('DAY', NOW() - INTERVAL '7' Day)
```

Comment:
We can use `incremental_predicate` for incremental filters:
```suggestion
	WHERE {{incremental_predicate('evt_block_time')}}
```